tests/unit/plugins/modules/net_tools/test_dnsimple_info.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+from ansible_collections.community.general.plugins.modules.net_tools import dnsimple_info
+from ansible_collections.community.general.tests.unit.plugins.modules.utils import AnsibleFailJson, ModuleTestCase, set_module_args
+from ansible_collections.community.general.tests.unit.compat.mock import patch
+import pytest
+
+dnsimple = pytest.importorskip('dnsimple_info')
+
+
+class TestDNSimple_Info(ModuleTestCase):
+    """Main class for testing dnsimple module."""
+
+    def setUp(self):
+        """Setup."""
+        super(TestDNSimple_Info, self).setUp()
+        self.module = dnsimple_info
+
+    def tearDown(self):
+        """Teardown."""
+        super(TestDNSimple_Info, self).tearDown()
+
+    def test_with_no_parameters(self):
+        """Failure must occurs when all parameters are missing"""
+        with self.assertRaises(AnsibleFailJson):
+            set_module_args({})
+            self.module.main()
+
+    @patch('dnsimple_info.main')
+    def test_only_key_and_account(self, mock_main):
+        """If we have key and id we should return domains,
+        ID in return should match supplied ID"""
+        mock_main.return_value.exit_json = {
```

Comment:
Seriously, if you mock the whole `main` function, you are testing nothing. You need to run the regular main function and mock the API calls.

Reply:
I'm not a developer and really this is my first project like this, and first PR, I do appreciate the advice, I guess I didn't really get the point of unit tests when we're just going to be supplying the return values anyway, but I have never read about them until they were suggested here.
I think a good approach would be to instead mock the function that makes the request, then like you said execute the main function.
If that sounds like a good approach I will give it a shot tomorrow, thanks!

Reply:
The best way would be to mock HTTP requests:
```.py
from httmock import response
from httmock import urlmatch


@urlmatch(scheme="http", netloc="localhost", path="/api/v4/users", method="get")
def resp_find_user(url, request):
    headers = {'content-type': 'application/json'}
    content = ('[{"id": 1, "username": "john_smith", "name": "John Smith", "state": "active",'
               '"avatar_url": "http://localhost:3000/uploads/user/avatar/1/cd8.jpeg",'
               '"web_url": "http://localhost:3000/john_smith"}, {"id": 2,'
               '"username": "jack_smith", "name": "Jack Smith", "state": "blocked",'
               '"avatar_url": "http://gravatar.com/../e32131cd8.jpeg",'
               '"web_url": "http://localhost:3000/jack_smith"}]')
    content = content.encode("utf-8")
    return response(200, content, headers, None, 5, request)
```
Then in the test you can simply add
```
    @with_httmock(resp_find_user)
```
to the test function, and in that function every HTTP GET call to the path `/api/v4/users` will get mocked by the above function.